mapper/thin_edge_json/src/json.rs

```diff
@@ -282,13 +276,109 @@ impl ThinEdgeJsonError {
 
 mod tests {
     use super::*;
+    use crate::measurement::*;
+    use assert_matches::*;
 
     fn test_timestamp() -> DateTime<FixedOffset> {
         FixedOffset::east(5 * 3600)
             .ymd(2021, 04, 08)
             .and_hms(0, 0, 0)
     }
 
+    #[test]
+    fn invalid_time_stamp() {
+        let input = r#"{
+            "time" : "2013-06-2217:03:14.000658767+02:00"
+        }"#;
+        let expected_error = r#"Invalid ISO8601 timestamp (expected YYYY-MM-DDThh:mm:ss.sss.±hh:mm): "2013-06-2217:03:14.000658767+02:00": input contains invalid characters"#;
+        let output_err = ThinEdgeJson::from_str(input, test_timestamp()).unwrap_err();
+        assert_eq!(output_err.to_string(), expected_error);
+    }
+
+    #[test]
+    fn test_parse_from_rfc3339_empty_timestamp() {
+        //no time stamp
+        assert_matches!(
+            ThinEdgeJson::parse_from_rfc3339("...."),
+            Err(ThinEdgeJsonError::InvalidTimestamp { .. })
+        );
+    }
+    #[test]
+    fn test_parse_from_rfc3339_timestamp_with_seconds() {
+        //no subseconds

```

Comment:
don't use comments here. the "no subseconds" should really be part of the test function name.

```rust
fn test_parse_from_rfc3339_timestamp_with_seconds_but_no_subseconds
```